src/router.rs

```diff
@@ -25,9 +25,14 @@ impl<Data> Router<Data> {
         path: &'a str,
         method: &http::Method,
     ) -> Option<(&'a BoxedEndpoint<Data>, RouteMatch<'a>)> {
-        self.table
-            .route(path)
-            .and_then(|(r, p)| Some((r.endpoints.get(method)?, p)))
+        let route = self.table.route(path)?;
```

Comment:
Maybe do something like `let (route, route_match) = self.table.route(path)?;` here? (I am not sure what could be the best names here though)

Then later you don't have to use `route.0` and `route.1`?

Reply:
Agreed, that'd be a nice code cleanup. Other than that, looks good to go!

Reply:
I am new to Rust as well, might not have the best opinion here but I simply find `tuple.0, tuple.1` difficult to read as they look like the tuple type but it could be completely a different type. Just from `tuple.0` I don't know the name of it and I don't know the type of it, without looking for the definition of `tuple` somewhere else.  I guess if you use them only once or twice and they are very close to where `tuple` is defined/created, then maybe it is fine. Especially if you only have one element in the tuple (like a wrap of some other type) then using `tuple.0` around is probably okay. Otherwise give them names might help a bit in most cases. Just my opinion :P